src/openrct2/actions/LandSmoothAction.hpp

```diff
@@ -368,55 +368,83 @@ DEFINE_GAME_ACTION(LandSmoothAction, GAME_COMMAND_EDIT_LAND_SMOOTH, GameActionRe
                 // Smooth the 4 corners
                 { // top-left
                     auto surfaceElement = map_get_surface_element_at({ validRange.GetLeft(), validRange.GetTop() });
-                    int32_t z = std::clamp((uint8_t)tile_element_get_corner_height(surfaceElement, 2), minHeight, maxHeight);
-                    res->Cost += SmoothLandRowByCorner(
-                        isExecuting, { validRange.GetLeft(), validRange.GetTop() }, z, -32, -32, 0, 2);
+                    if (surfaceElement != nullptr)
```

Comment:
This will likely introduce a different bug if the surface element can not be found, this can be triggered with the Tile Inspector by deleting one of the surface elements and then using the Land tool afterwards. Could you perhaps check this with a debug build?

Reply:
I don't really know how  to do this :(

Reply:
![image](https://user-images.githubusercontent.com/6443427/66913577-4a097e80-efeb-11e9-972d-0b915b8144b0.png)
So I deleted the middle element from the surface there and it starts flickering on display (showing a tall terrain at times or showing none at all). Trying to elevate the terrain works, but trying to lower it sometimes yield the error shown above.

Reply:
This is to be somewhat expected if you erase the surface element, the rendering artifact is another issue on its own. I recall that std::clamp did not like the inverted min/max values, are you using a debug build?

Reply:
Yup, to make sure I just reran after this:
```
set platform=x64
msbuild openrct2.proj /t:clean
msbuild openrct2.proj /t:build /p:configuration=debug
```
And the results were the same

Reply:
Have you tried smooth land?

Reply:
Actually seemed to work as well
![image](https://user-images.githubusercontent.com/6443427/66968877-76f67980-f05d-11e9-92f0-75faa3584fcc.png)

Reply:
I think this is fine.